src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.IDictionary.cs

```diff
@@ -34,6 +34,43 @@ public void Add(string propertyName, JsonNode? value)
             value?.AssignParent(this);
         }
 
+        /// <summary>
+        ///   Adds an element with the provided name and value to the <see cref="JsonObject"/>, if a property named <paramref name="propertyName"/> doesn't already exist..
+        /// </summary>
+        /// <param name="propertyName">The property name of the element to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="propertyName"/> is null.</exception>
+        /// <returns>
+        ///   <see langword="true"/> if the property didn't exist and the element was added; otherwise, <see langword="false"/>.
+        /// </returns>
+        public bool TryAdd(string propertyName, JsonNode? value) => TryAdd(propertyName, value, out _);
+
+        /// <summary>
+        ///   Adds an element with the provided name and value to the <see cref="JsonObject"/>, if a property named <paramref name="propertyName"/> doesn't already exist.
+        /// </summary>
+        /// <param name="propertyName">The property name of the element to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <param name="index">The index of the added or existing <paramref name="propertyName"/>. This is always a valid index into the <see cref="JsonObject"/>.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="propertyName"/> is null.</exception>
+        /// <returns>
+        ///   <see langword="true"/> if the property didn't exist and the element was added; otherwise, <see langword="false"/>.
+        /// </returns>
+        public bool TryAdd(string propertyName, JsonNode? value, out int index)
+        {
+            if (propertyName is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(propertyName));
+            }
+
+            var success = Dictionary.TryAdd(propertyName, value, out index);
```

Comment:
This is probably failing to build in .NET 9 since that version of OrderedDictionary doesn't have the overload that exposes the index. I would recommend emulating the behavior using a second lookup via `IndexOf` in .NET 9 targets.